ansible/pbTestScripts/testScript.sh

```diff
@@ -4,29 +4,104 @@ set -u
 branchName='NULL'
 folderName=''
 gitURL=''
+vagrantOS=''
+retainVM=false
+testNativeBuild=false
 
-# Takes all arguments from the script
+# Takes all arguments from the script, and determines options
 processArgs()
 {
-	if [ $# -lt 2 ]; then
-		printf "\nScript takes 2 input arguments\n"
-		printf "Github URL, then y/n to retain VMs\n\n"
-		exit 1
+	while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+		local opt="$1";
+		shift;
+		case "$opt" in
+			"--Vagrantfile" | "-v" )
+				vagrantOS="$1"; shift;;
+			"--all" | "-a" )
+				vagrantOS="all";;
+			"--build" | "-b" )
+				testNativeBuild=true;;
+			"--retainVM" | "-r" )
+				retainVM=true;;
+			"--URL" | "-u" )
+				gitURL="$1"; shift;;
+			"--help" | "-h" )
+				usage; exit 0;;
+			*) echo >&2 "Invalid option: ${opt}"; echo "This option was unrecognised."; usage; exit 1;;
+		esac
+	done
+}
+
+usage()
+{
+	echo
+	echo "Usage: ./testScript.sh			--vagrantfile | -v <OS_Version>		Specifies which OS the VM is
+					--all | -a 				Builds and tests playbook through every OS
+					--retainVM | -r				Option to retain the VM once building them
+					--build | -b				Option to enable testing a native build on the VM
+					--URL | -u <GitURL>			The URL of the git repository
+					--help | -h				Displays this help message"
+}
+
+defaultVars()
+{
+	#Sets WORKSPACE to home if WORKSPACE is empty or undefined. 
+	if [ ! -n "${WORKSPACE:-}" ]; then
```

Comment:
What's this construct doing that wouldn't be covered by "if [ -z "$WORKSPACE" ]?

Reply:
`if [ -z "$WORKSPACE" ]` will error if $WORKSPACE is undefined

Reply:
It's the `set -u` that's causing that